xbmc/cores/AudioEngine/Encoders/AEEncoderFFmpeg.cpp

```diff
@@ -290,40 +290,39 @@ int CAEEncoderFFmpeg::Encode(uint8_t *in, int in_size, uint8_t *out, int out_siz
 
     avcodec_fill_audio_frame(frame, channelNum, m_CodecCtx->sample_fmt, in, in_size, 0);
 
-    pkt->size = out_size;
-    pkt->data = out;
-
     /* encode it */
     err = avcodec_send_frame(m_CodecCtx, frame);
     if (err < 0)
       throw FFMpegException("Error sending a frame for encoding (error '{}')",
                             FFMpegErrorToString(err));
 
-    while (err >= 0)
+    err = avcodec_receive_packet(m_CodecCtx, pkt);
+    // err < 0 - we cannot cope with it
+    // err is EAGAIN or EOF - return to caller as well
+    if (err >= 0 && err != AVERROR(EAGAIN) && err != AVERROR_EOF)
```

Comment:
According to your comments: if (err >= 0) is enough as EAGAIN is a negative value.

Reply:
Yep :) Checked that, both values are negative!